checker/src/main/java/org/checkerframework/checker/determinism/DeterminismAnnotatedTypeFactory.java

```diff
@@ -433,11 +483,11 @@ void recursiveDefaultArrayComponentType(
     }
 
     /**
-     * If {@code type} is a collection type that is not explicitly annotated, defaults all its
+     * If {@code type} is a Collection type that is not explicitly annotated, defaults all its
      * nested component types as {@code annotation}.
      */
     void defaultCollectionComponentType(AnnotatedTypeMirror type, AnnotationMirror annotation) {
-        if (isCollection(type)
+        if ((isCollection(type) || isMap(type))
```

Comment:
I have added the check for `Iterators` here. I haven't called `mayBeOrderNonDet` because that method also checks for array types and casting them to `AnnotatedDeclaredType` is causing a class cast exception. I agree that we should refactor this, but may be later.

Reply:
Ok. Looks like you fixed it in this method but also need to change it in `recursiveDefaultCollectionComponentType`.

Reply:
Great catch!